internal/build/phase_config_provider.go

```diff
@@ -108,11 +111,33 @@ func WithBinds(binds ...string) PhaseConfigProviderOperation {
 func WithDaemonAccess() PhaseConfigProviderOperation {
 	return func(provider *PhaseConfigProvider) {
 		WithRoot()(provider)
+
 		bind := "/var/run/docker.sock:/var/run/docker.sock"
+
+		if dockerHost, ok := os.LookupEnv("DOCKER_HOST"); ok {
+			if strings.HasPrefix(dockerHost, "unix://") {
+				bind = fmt.Sprintf("%s:/var/run/docker.sock", strings.TrimPrefix(dockerHost, "unix://"))
+			}
+			if strings.HasPrefix(dockerHost, "tcp://") {
+				bind = ""
+				if runtime.GOOS == "linux" {
+					provider.hostConf.NetworkMode = "host"
+					provider.ctrConf.Env = append(provider.ctrConf.Env, fmt.Sprintf("DOCKER_HOST=%s", dockerHost))
+				} else {
+					w := provider.errorWriter
+					if w == nil {
+						w = os.Stderr
+					}
+					fmt.Fprintln(w, "Don't know how to expose Docker API via TCP to the container.")
```

Comment:
Writing directly to the errorWriter would bypass our typical styles. We should probably look for another way of sending that error message that would be a more seamless fit with our normal practices.